### ISSUE TITLE
Progress bar shows empty `v_num` label if running with `fast_dev_run` enabled

### ISSUE THREAD

**Issue by awaelchli** [Lightning-AI/lightning#18489]
### Bug description

When running with fast-dev-run enabled, the Trainer uses a DummyLogger that has its version defined as the empty string. This results in the progress bar showing an empty `v_num` label:

```
Epoch 0: 100%|██████████| 1/1 [00:00<00:00, 25.93it/s, v_num=]
```

This looks awkward. 

### What version are you seeing the problem on?

v2.0, master

### How to reproduce the bug

```python
import torch
from torch.utils.data import DataLoader, Dataset

from lightning.pytorch import LightningModule, Trainer


class RandomDataset(Dataset):
    def __init__(self, size, length):
        self.len = length
        self.data = torch.randn(length, size)

    def __getitem__(self, index):
        return self.data[index]

    def __len__(self):
        return self.len


class BoringModel(LightningModule):
    def __init__(self):
        super().__init__()
        self.layer = torch.nn.Linear(32, 2)

    def training_step(self, batch, batch_idx):
        return self.layer(batch).sum()

    def configure_optimizers(self):
        return torch.optim.SGD(self.layer.parameters(), lr=0.1)


train_data = DataLoader(RandomDataset(32, 64), batch_size=2)
model = BoringModel()
trainer = Trainer(fast_dev_run=True)
trainer.fit(model, train_data)
```

### Error messages and logs

No error.

### Environment

Lightning 2.1.0dev
PyTorch 2.0
Python 3.10

cc @awaelchli
